examples/golang/main.go

```diff
@@ -229,13 +249,434 @@ func getProductsByCategory(svc *dynamodb.DynamoDB) {
 
+
+func updateCustomerDetails(svc *dynamodb.DynamoDB) {
+	_, err := svc.UpdateItem(&dynamodb.UpdateItemInput{
+		TableName: aws.String("Customer_Order"),
+		Key: map[string]*dynamodb.AttributeValue{
+    			"PK": {
+    				S: aws.String("CUST#0000000000"),
+    			},
+    			"SK": {
+    				S: aws.String("EMAIL#homer@email.com"),
+    			},
+    		},
+    ExpressionAttributeValues: map[string]*dynamodb.AttributeValue{
+            ":number_of_items": {
+                S: aws.String("10"),
+            },
+        },
+    ReturnValues: aws.String("UPDATED_NEW"),
+    UpdateExpression: aws.String("seT number_of_items = :number_of_items"),
+	})
+
+	if err != nil {
+		fmt.Println(err.Error())
+	}
+	getCustomerContactDetails(svc,"CUST#0000000000","EMAIL#homer@email.com")
+
+
+}
+
+
+func addNewCustomer(svc *dynamodb.DynamoDB) {
+
+  type ItemToPut struct {
+      PK                 string
+      SK                 string
+      Addresses          interface{} `dynamodbav:"customer_addresses"`
+      Email              string      `dynamodbav:"customer_email"`
+      Fname              string      `dynamodbav:"customer_fname"`
+      Id                 string      `dynamodbav:"customer_id"`
+      Lname              string      `dynamodbav:"customer_lname"`
+
+  }
+
+  itemtoput := ItemToPut{
+      PK:  "CUST#9922885566",
+      SK:  "EMAIL#mosby@gmail.com",
+      Addresses:  "{Shipping:  Maclarens pub, 240 W 55th St, New York, NY}",
+      Email:  "mosby@gmail.com",
+      Fname:  "Ted",
+      Id:  "9922885566",
+      Lname:  "Mosby",
+  }
+
+  av, err := dynamodbattribute.MarshalMap(itemtoput)
+  if err != nil {
+      fmt.Println(err.Error())
+  }
+  fmt.Println("Adding new customer with PK=CUST#9922885566 and SK=EMAIL#mosby@gmail.com")
+	_, error := svc.PutItem(&dynamodb.PutItemInput{
+		TableName: aws.String("Customer_Order"),
+		Item: av,
+	})
+
+	if error != nil {
+		fmt.Println(error.Error())
+	}
+  fmt.Println("Verifying the customer has been added")
+	getCustomerContactDetails(svc,"CUST#9922885566","EMAIL#mosby@gmail.com")
+}
+
+func deleteCustomer(svc *dynamodb.DynamoDB) {
+
+  fmt.Println("Deleting the customer with PK=CUST#9922885566 and SK=EMAIL#mosby@gmail.com")
+	_, err := svc.DeleteItem(&dynamodb.DeleteItemInput{
+		TableName: aws.String("Customer_Order"),
+		Key: map[string]*dynamodb.AttributeValue{
+			"PK": {
+				S: aws.String("CUST#9922885566"),
+			},
+			"SK": {
+				S: aws.String("EMAIL#mosby@gmail.com"),
+			},
+		},
+	})
+
+	if err != nil {
+      fmt.Println(err.Error())
+  }  else {
+      fmt.Println("Deleted the customer with PK=CUST#9922885566 and SK=EMAIL#mosby@gmail.com")
+  }
+
+  fmt.Println("Verifying the customer has been deleted")
+  getCustomerContactDetails(svc,"CUST#9922885566","EMAIL#mosby@gmail.com")
+
+}
+
+func getCustomerWithSameId(svc *dynamodb.DynamoDB) {
+
+  fmt.Println("Running a scan operation to find customers with Id 0000000000 who has number_of_items greater than 3")
+  type Item struct {
```

Comment:
Why not use the existing types or make a new `CustomerSummary` type or something like that.